cedar-policy-validator/src/human_schema/fmt.rs

```diff
@@ -183,6 +184,23 @@ impl NameCollisionsError {
 }
 
 /// Convert a [`SchemaFragment`] to a string containing human schema syntax
+///
+/// REVIEW: The existing code throws an error if any fully-qualified name in a
+/// non-empty namespace is a valid common type and also a valid entity type.
+/// 1) I think this check is more conservative than necessary? Schemas are
+/// allowed to shadow an entity type with a common type declaration in the same
+/// namespace; see RFCs 24 and 70. What the human syntax can't express is if, in
+/// that situation, we then specifically refer to the shadowed entity type name.
+/// But it's harder to walk all type references than it is to walk all type
+/// declarations, so perhaps the conservative code here is fine.
+/// 2) At the same time, this check might miss some cases that are actually
```

Comment:
>  When converted to human syntax, the reference will resolve to the common type declaration in the current namespace instead.

Sounds like a bug?

Reply:
While trying to reproduce I wondered, do we have a fuzz target that generates a JSON schema, converts to natural, and converts back?

Reply:
That should be https://github.com/cedar-policy/cedar-spec/blame/main/cedar-drt/fuzz/fuzz_targets/convert-schema-json-to-human.rs, but that's using arbitrary strings, so not the best coverage

Reply:
Filed the bug as #1063.  It affects at minimum 3.2.1, I would guess all versions back to 3.1.0 which introduced the human/natural schema syntax.